src/content/fr/2019/caching.md

```diff
@@ -0,0 +1,691 @@
+
+**Mettre en cache aussi longtemps que possible.** La durée de mise en cache d'une ressource dépend fortement de sa sensibilité et de son contenu. Une ressource JavaScript versionnée peut être mise en cache pendant très longtemps, alors qu'une ressource non versionnée peut nécessiter une durée de cache plus courte afin de garantir aux utilisateurs de disposer des données a jour.
+
+**Cachez le plus près possible des utilisateurs finaux.** Une mise en cache proche des utilisateurs réduit les temps de téléchargement en réduisant les latences réseau. Par exemple, pour une ressource mise en cache sur le navigateur de l'utilisateur, la requête ne sera jamais envoyée sur le réseau et le temps de téléchargement sera aussi rapide que les I/O de la machine. Pour les premiers visiteurs, ou les visiteurs qui n'ont pas encore leurs données cachées, un CDN est généralement la prochaine localisation d'une ressource cachée. Dans la plupart des cas, il sera plus rapide de récupérer une ressource à partir d'un cache local ou d'un CDN que sur le serveur d'origine.
+
+Les architectures Web impliquent généralement [une mise en cache en plusieurs niveaux](https://blog.yoav.ws/tale-of-four-caches/). Par exemple une requête HTTP peut être mise en cache de différentes manière&nbsp;:
+
+*   dans le cache du navigateur&nbsp;;
+*   dans le cache d'un <i lang="en">service worker</i> dans le navigateur&nbsp;;
+*   dans une passerelle partagée&nbsp;;
+*   au niveau des CDN, qui offrent la possibilité de mettre en cache à proximité des utilisateurs&nbsp;;
+*   dans un proxy de cache en amont des applications pour réduire la charge sur les serveurs back-end&nbsp;;
+*   au niveau de l'application et de la base de données.
+
+Ce chapitre explique comment les ressources sont mises en cache dans les navigateurs Web.
+
+## Présentation de la mise en cache HTTP
+
+Pour qu'un client HTTP mette en cache une ressource, il doit répondre a deux questions&nbsp;:
+
+*   "Combien de temps dois-je mettre en cache&nbsp;?"
+*   "Comment puis-je valider que le contenu est encore frais&nbsp;?"
+
+Lorsqu'un navigateur Web envoie une réponse à un client, il inclut généralement dans sa réponse des en-têtes qui indiquent si la ressource peut être mise en cache, pour combien de temps et quel est son âge. La RFC 7234 traite plus en détail de ce point dans la section [4.2 (Freshness)](https://tools.ietf.org/html/rfc7234#section-4.2) et [4.3 (Validation)](https://tools.ietf.org/html/rfc7234#section-4.3).
+
+Les en-têtes de réponse HTTP généralement utilisées pour transmettre la durée de vie sont&nbsp;:
+
+*   `Cache-Control` vous permet de configurer la durée de vie du cache (c'est-à-dire sa durée de validité).
```

Comment:
Do we need to state that all the directives are English language terms or is that overkill? If so need to check this doesn't mess up the markdown->HTML script.

```suggestion
*   <i lang="en">`Cache-Control`<span> vous permet de configurer la durée de vie du cache (c'est-à-dire sa durée de validité).
```